### DISCUSSION TITLE
Clash with markdown linting rules for lists

### DISCUSSION THREAD

**Discussion by Prodeguerriero** [davidlday/vscode-languagetool-linter#765] [Q&A]
Hi all,

I'd like to hear your opinion on something. Please look at the markdown snippet below:

```markdown
1. Use `stow` to create the symlinks: `stow . -t ~/ --simulate --verbose`
   - `stow .` tells *stow* that we want every file in the current folder to be targeted by a symlinks.
   - `-t ~/` sets *$HOME* as the target directory, where the symlinks will be created.
   NOTE: If we don't specify a target directory, *stow* uses the parent folder of the stow directory as the target directory.
   - `--simulate --verbose` provides a verbose output of a dry run. No changes will be made to the file system.
````

LanguageTool complains about two broken rules:
- UPPERCASE_SENTENCE_START twice, for the lines starting with `stow .` and `--simulate --verbose`
- WHITESPACE_RULE for the line starting with "NOTE"

I don't understand why the first rule applies to the two lines that begin with `-` but not the one that begins with `stow`. But anyway, are these conflicts that you think are worth trying to resolve somehow?

Cheers

**Comment by davidlday** [comment]
Great questions. Here's what I see:
![image](https://github.com/user-attachments/assets/4f41bd26-6c1b-491e-8909-f9d9dfbd5218)

The UPPERCASE_SENTENCE_START isn't applied to the first `stow .` in mine because there's no preceding period on the `1.` line, so it doesn't see a new sentence. The rule is applied to the other two lines because the previous lines end in a period. I typically exclude ending punctuation in lists out of habit, so I don't usually see this problem.

The WHITESPACE_RULE isn't showing for me because my markdown formatter (prettier) automatically joined it with the preceding line. I believe in Markdown, if you want to inject a line like that within a list, it should have a blank line before and after, like this:
![image](https://github.com/user-attachments/assets/15c5632d-50e9-48f2-b007-ea8f05fc5f0f)

If I clean it up per my formatter and habit, I get:
![image](https://github.com/user-attachments/assets/4ed947b5-ea11-4dd8-892c-74ab5f781c72)

The last bullet point still gets the UPPERCASE_SENTENCE_START because of the NOTE in the list. There's feature for temporarily ignoring a rule using an inline comment, (see: https://github.com/davidlday/vscode-languagetool-linter?tab=readme-ov-file#ignore-rules-inline), but I don't seem to be able to get it working in this case. Will look into it more.

Hope this helps!

**Reply by Prodeguerriero** [reply]
Hi @davidlday,

Thanks for the reply, this helps a lot. For now, I have removed the periods at the end of the list items, which has solved one of the problems.

Regarding the "NOTE" line, we seem to be doing two different things.

When one wants to create a line break, [one needs to add two spaces to the previous line, then press enter](https://www.markdownguide.org/basic-syntax/#line-breaks).
Adding the empty extra line for padding is instead [creating a new paragraph](https://spec.commonmark.org/0.31.2/#example-309).
You can see the different HTML tags being generated in the example below:
![image](https://github.com/user-attachments/assets/99fb535f-e8eb-48a6-b455-a90854b3e3e3)

It was a fun rabbit hole to go down into. 😛 Besides, I just realized I had messed up my formatting, which generated the issue. So this is all sorted.

Regarding the ignore rules: [I left a comment to another Q&A thread](https://github.com/davidlday/vscode-languagetool-linter/discussions/742#discussioncomment-10671801), where I asked if there's a way for the user to define an ignore rule for the entire document. Before issuing a feature request, I just wanted to make sure I am not missing an existing feature. 😄 

Cheers!
